upup/pkg/fi/cloudup/awstasks/targetgroup.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awstasks
+
+import (
+	"fmt"
+
+	"k8s.io/klog/v2"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+	"k8s.io/kops/upup/pkg/fi/cloudup/cloudformation"
+	"k8s.io/kops/upup/pkg/fi/cloudup/terraform"
+)
+
+// TargetGroup manages an targetgroup used for an ALB/NLB.
+// +kops:fitask
+type TargetGroup struct {
+	Name      *string
+	Lifecycle *fi.Lifecycle
+
+	// ARN is the Amazon Resource Name for the Target Group
+	ARN *string
+
+	// Shared is set if this is an external LB (one we don't create or own)
+	Shared *bool
+}
```

Comment:
I created the `TargetGroup` struct mostly as a skeleton struct so that if someone decides to implement TargetGroup for uses in other places (such as API) they won't then have to figure out how to convert a list of string in `AutoscalingGroup` to an actual struct.